api/database_cluster_backup_test.go

```diff
@@ -0,0 +1,226 @@
+// everest
+// Copyright (C) 2023 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	everestv1alpha1 "github.com/percona/everest-operator/api/v1alpha1"
+	"github.com/percona/everest/pkg/kubernetes"
+	"github.com/percona/everest/pkg/rbac"
+	"github.com/percona/everest/pkg/rbac/mocks"
+	"github.com/percona/everest/pkg/session"
+)
+
+type enforceStrings struct{ subject, resource, action, object string }
+
+func TestListDBCluster_permissions(t *testing.T) {
+	t.Parallel()
+
+	err := everestv1alpha1.SchemeBuilder.AddToScheme(scheme.Scheme)
+	require.NoError(t, err)
+	metav1.AddToGroupVersion(scheme.Scheme, everestv1alpha1.GroupVersion)
+
+	l := zap.NewNop().Sugar()
+
+	testCases := []struct {
+		name          string
+		apiCall       func(echo.Context, *EverestServer) error
+		body          any
+		httpMethod    string
+		kubeClient    func() kubernetes.KubernetesConnector
+		proxyResponse any
+
+		expectedEnforce []enforceStrings
+	}{
+		{
+			name:       "List DB cluster backups",
+			httpMethod: http.MethodGet,
+			apiCall: func(ctx echo.Context, everest *EverestServer) error {
+				return everest.ListDatabaseClusterBackups(ctx, "ns", "my-name")
+			},
+			proxyResponse: &everestv1alpha1.DatabaseClusterBackupList{
+				TypeMeta: metav1.TypeMeta{
+					Kind: "database-cluster-backup-list",
+				},
+				Items: []everestv1alpha1.DatabaseClusterBackup{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "name1",
+							Namespace: "ns",
+						},
+						Spec: everestv1alpha1.DatabaseClusterBackupSpec{
+							DBClusterName: "db-cluster-name",
+						},
+					},
+				},
+			},
+
+			expectedEnforce: []enforceStrings{
+				{"alice", rbac.ResourceBackupStorages, rbac.ActionRead, "ns/name1"},
```

Comment:
I think these tests just helped us find a bug :sweat_smile: 
The enforce for ResourceBackupStorages not be `ns/name1`. Instead it should be `ns/backup-storage-name` that is part of the Spec like so:

```suggestion
						Spec: everestv1alpha1.DatabaseClusterBackupSpec{
							DBClusterName: "db-cluster-name",
							BackupStorageName: "backup-storage-name",
						},
					},
				},
			},

			expectedEnforce: []enforceStrings{
				{"alice", rbac.ResourceBackupStorages, rbac.ActionRead, "ns/backup-storage-name"},
```

The bug is [here](https://github.com/percona/everest/blob/316f08d9c843d2b63f455a524238b6097e12e62f/api/database_cluster_backup.go#L100). It should be `dbbackup.Spec.BackupStorageName` instead of `dbbackup.GetName()`

Reply:
The other functions have the same bug :see_no_evil: